src/pages/Swap/LimitOrderBox/components/LimitOrderForm/LimitOrderForm.tsx

```diff
@@ -0,0 +1,513 @@
+        sellTokenAmount?.currency?.decimals
+      ).toString()
+      const expiresAt = dayjs().add(expiresIn, expiresInUnit).unix()
+      const sellCurrencyMaxAmount = maxAmountSpend(sellCurrencyBalance, chainId)
+
+      checkMaxOrderAmount(
+        limitOrder,
+        setIsPossibleToOrder,
+        setLimitOrder,
+        amountWei,
+        expiresAt,
+        sellTokenAmount,
+        sellCurrencyMaxAmount,
+        chainId,
+        provider
+      )
+    }, 500)
+    return () => clearTimeout(timeoutId)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [formattedSellAmount, formattedBuyAmount, newSellAmount])
+
+  /**
+   * Aggregate the sell currency amount state variables into a single value
+   */
+  const handleSellCurrencyAmountChange = ({
+    amountWei,
+    currency,
+    amountFormatted,
+    updatedLimitOrder = limitOrder,
+  }: HandleCurrencyAmountChangeParams) => {
+    const limitPriceFloat = parseFloat(formattedLimitPrice)
+
+    // Construct a new token amount and format it
+    const newLimitOrder = {
+      ...updatedLimitOrder,
+      sellAmount: amountWei,
+      sellToken: currency.address as string,
+    }
+    // Update the buy currency amount if the user has selected a token
+    // Update relevant state variables
+    const nextSellTokenAmount = new TokenAmount(currency as Token, amountWei)
+    const nextSellAmountFloat = parseFloat(amountFormatted)
+
+    setFormattedSellAmount(amountFormatted) // update the token amount input
+
+    let nextBuyAmountFloat = 0
+    // Compute the buy amount based on the new sell amount
+
+    if (limitOrder.kind === LimitOrderKind.SELL) {
+      nextBuyAmountFloat = nextSellAmountFloat * limitPriceFloat
+    } else {
+      nextBuyAmountFloat = nextSellAmountFloat / limitPriceFloat
+    }
+    if ((Number(amountWei) ?? 0) === 0) {
+      setIsPossibleToOrder({ value: 0, status: true })
+    }
+
+    // Format the buy amount
+    // Update buy amount state variables
+    setBuyTokenAmount(
+      new TokenAmount(
+        buyTokenAmount.currency as Token,
+        parseUnits(nextBuyAmountFloat.toFixed(6), buyTokenAmount.currency.decimals).toString()
+      )
+    )
+    setFormattedBuyAmount(nextBuyAmountFloat.toString()) // update the token amount input
+    setSellTokenAmount(nextSellTokenAmount)
+    // Re-compute the limit order boy
+    setLimitOrder(newLimitOrder)
+  }
+
+  /**
+   * Aggregate the buy currency amount state variables into a single value
+   */
+  const handleBuyCurrencyAmountChange = ({
+    amountWei,
+    currency,
+    amountFormatted,
+    updatedLimitOrder = limitOrder,
+  }: HandleCurrencyAmountChangeParams) => {
+    // Construct a new token amount and format it
+
+    const newLimitOrder = {
+      ...updatedLimitOrder,
+      buyAmount: amountWei,
+      buyToken: currency.address as string,
+    }
+    const nextBuyAmountFormatted = amountFormatted // When the limit price is empty, set the limit price to 0
```

Comment:
This comment seems out of context. There is no code setting price to 0.

Reply:
@berteotti You can fix all the issues you see. I think you can now own and complete it.